### PR TITLE
New specialized client for the GNSS end-point on scihub

### PR DESCRIPTION
**Pull request by avalentino**
This PR provides a new, specialized, API client for the GNSS end-point on scihub.

~**NOTE**: this PR is based on the ``feature/gha`` that implements CI testing with GitHub Actions.~

Closes: #436.

### CODE REVIEW

**Comment by j08lue** [comment]
Is this ready for review? Or do you want to wait until the GH Actions PR is merged?

**Comment by avalentino** [comment]
> Is this ready for review? Or do you want to wait until the GH Actions PR is merged?

Probably it would make sense to wait the PR for GHA is merged.
Please note that at the moment I have not added any new test.
I'm considering to add some basic one but probably I will need you support for that (I cannot make tests on Windows).

Also we can discuss some additional and more advanced features targeting a new PR.

File: sentinelsat/scripts/cli.py
Comment (valgur) [inline sentinelsat/scripts/cli.py:275]: The `sentinel` parameter might not have been set.

Comment (avalentino) [reply sentinelsat/scripts/cli.py:273]: done, thanks

File: docs/cli.rst
Comment (valgur) [inline docs/cli.rst:204]:
```suggestion
    Query orbit products form the GNSS end-point (https://scihub.copernicus.eu/gnss).
```

File: README.rst
Comment (valgur) [inline README.rst:244]:
```suggestion
     - Query orbit products form the GNSS end-point (https://scihub.copernicus.eu/gnss).
```

File: CHANGELOG.rst
Comment (valgur) [inline CHANGELOG.rst:21]: Please add a changelog entry under the "CLI changes" section.

**Review by valgur — CHANGES_REQUESTED** [review]
So this PR really boils down to simply providing a slightly more convenient CLI interface by adding a shortcut for
```
sentinelsat --url https://scihub.copernicus.eu/gnss -u gnssguest -p gnssguest ...
```
with
```
sentinelsat --gnss ...
```

File: sentinelsat/scripts/cli.py
Comment (avalentino) [reply sentinelsat/scripts/cli.py:261]: Uhm, the point is that I don't want override inputs provided by the user.
Right now 
```
url = "https://scihub.copernicus.eu/gnss/"
user = "gnssguest"
password = "gnssguest"
```
are the only possible parameters things could change in the future.

File: sentinelsat/scripts/cli.py
Comment (avalentino) [reply sentinelsat/scripts/cli.py:271]: E.g. `AUX_POEORB` is a valid `producttype` both for S1 and S3, so one may need to filter also the platform for which the orbit is requested.

File: README.rst
Comment (avalentino) [reply README.rst:244]: +1

File: sentinelsat/gnss.py
Comment (avalentino) [reply sentinelsat/gnss.py:8]: Yes I agree.
The original class was written to overcome some limitations in the implementation of the base class w.r.t. the GNSS case.
Now it seems that all issues have been resolved or will be resolved soon (e.g. #378 and #472).
Thanks a lot :)

File: sentinelsat/scripts/cli.py
Comment (avalentino) [reply sentinelsat/scripts/cli.py:261]: Uhm, the point is that I don't want override inputs provided by the user.
Right now 
```
url = "https://scihub.copernicus.eu/gnss/"
user = "gnssguest"
password = "gnssguest"
```
are the only possible parameters things could change in the future.

File: docs/cli.rst
Comment (avalentino) [reply docs/cli.rst:204]: good catch!*

File: CHANGELOG.rst
Comment (avalentino) [reply CHANGELOG.rst:21]: OK

File: sentinelsat/scripts/cli.py
Comment (valgur) [reply sentinelsat/scripts/cli.py:261]: That's true, but in that case the user should use the equivalent of
```
sentinelsat --url https://scihub.copernicus.eu/gnss -u gnssguest -p gnssguest
```
instad of using `--gnss`.

File: sentinelsat/scripts/cli.py
Comment (valgur) [reply sentinelsat/scripts/cli.py:271]: Hmm... I did not remember that we have a separate option hierarchy logic in place here. In retrospect that does not seem like a particularly good idea to me. I would simplify this to just
```python
    if sentinel:
        search_kwargs["platformname"] = "Sentinel-" + sentinel

    if instrument:
        search_kwargs["instrumentshortname"] = instrument

    if producttype:
        search_kwargs["producttype"] = producttype
```
and get rid of [test_option_hierarchy](https://github.com/sentinelsat/sentinelsat/blob/main/tests/test_cli.py#L430-L451) (along with the corresponding cassette).

File: sentinelsat/scripts/cli.py
Comment (avalentino) [reply sentinelsat/scripts/cli.py:261]: looks fair

File: sentinelsat/scripts/cli.py
Comment (avalentino) [reply sentinelsat/scripts/cli.py]: done

**Review by valgur — APPROVED** [review]
Looks good! The only thing left is to merge with `main`. You will have to re-record the modified cassettes due to the POST->GET change.

**Comment by avalentino** [comment]
done

File: tests/test_cli.py
Comment (valgur) [inline tests/test_cli.py:145]: Looks like you accidentally duplicated this test.

**Review by valgur — CHANGES_REQUESTED** [review]

File: tests/test_cli.py
Comment (avalentino) [reply tests/test_cli.py:138]: ups, sorry
should be fixed now

**Pull request merged**
